src/client/keycode.cpp

```diff
@@ -221,122 +223,177 @@ static const struct table_key table[] = {
 	DEFINEKEY5("_")
 };
 
+static const table_key invalid_key = {"", irr::KEY_UNKNOWN, L'\0', ""};
+
 #undef N_
 
 
-static const table_key &lookup_keyname(const char *name)
+static const table_key &lookup_keychar(wchar_t Char)
 {
+	if (Char == L'\0')
+		return invalid_key;
+
 	for (const auto &table_key : table) {
-		if (strcmp(table_key.Name, name) == 0)
+		if (table_key.Char == Char)
 			return table_key;
 	}
 
-	throw UnknownKeycode(name);
+	auto newsym = wide_to_utf8(std::wstring_view(&Char, 1));
+	table_key new_key {newsym, irr::KEY_KEY_CODES_COUNT, Char, newsym};
```

Comment:
this needs some explanation

Reply:
this seems to imply we need to be able to return valid data for any wchar. why is that?

Reply:
This is mostly for convenience so that we have an entry to work with if the keychar is not listed in the lookup table.

For SDL devices this is also safe, since the only cases where we get invalid wchar values are
* From SDL, which we sanitize in `findCharToPassToIrrlicht`, or
* From user input (i.e. `minetest.conf`), in which case the keychar is then converted to the `SDL_SCANCODE_UNKNOWN`.